### DISCUSSION TITLE
RunnerSets. Their replicas and Runners' status

### DISCUSSION THREAD

**Discussion by sadovnikov** [actions/actions-runner-controller#2256] [Questions]
Hello, in order to start caching docker layers among a set of runners, I tried to move from RunnerDeployment to RunnerSet. That didn't really work for two reasons below.

I wonder why an increase of `replicas` in RunnerSets results in increasing the number of StatefulSets, instead of changing replicas of the existing STS. It seems to defeat the purpose of STS, because STSs (with new names) are kept on being re-created. Since STSs are re-created, the linked templated PVCs get removed too.

When RunnerDeployments are used, there is a Runner CR for every runner pod. The status of these CRs reflects the status of the runner (Pending, Idle, Running). How the runner status is being exposed when RunnerSets are used?

**Comment by harshaisgud** [comment]
Please check this out and see if it helps you.I am using snapshots to setup the cache.  https://github.com/actions/actions-runner-controller/discussions/2253
